### PR TITLE
feat(autocomplete): move harness out of experimental

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

**Comment by mmalerba** [comment]
CARETAKER NOTE: merge-safe, but may require BUILD file updates in g3

File: test/karma-system-config.js [hunk 1]
```diff
@@ -96,6 +96,8 @@ System.config({
     '@angular/cdk-experimental/scrolling': 'dist/packages/cdk-experimental/scrolling/index.js',
 
     '@angular/material/autocomplete': 'dist/packages/material/autocomplete/index.js',
+    '@angular/material/autocomplete/testing': 'dist/packages/material/autocomplete/testing/index.js',
+    '@angular/material/autocomplete/testing/shared.spec': 'dist/packages/material/autocomplete/testing/shared.spec.js',
```
Comment (devversion) [inline test/karma-system-config.js]: Ugh, not really nice that we have to add this for each harness we move, but yeah, I see why we need it.

File: src/material/autocomplete/testing/shared.spec.ts [hunk 2]
```diff
@@ -1,84 +1,62 @@
...
-
-    // TODO: enable after MDC autocomplete is implemented
-    // runTests();
+import {MatAutocompleteHarness} from '@angular/material/autocomplete/testing';
+
+/**
+ * Function that can be used to run the shared autocomplete harness tests for either the non-MDC or
+ * MDC based checkbox harness.
+ */
+export function runTests(
```
Comment (devversion) [inline src/material/autocomplete/testing/shared.spec.ts]: Should we name this `runHarnessTests`? Nothing from the package import implies that these are tests using the harnesses (i.e. `@angular/material/autocomplete/testing`)

File: src/material/autocomplete/testing/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    ),
+    module_name = "@angular/material/autocomplete/testing",
+    deps = [
+        "//src/cdk/coercion",
+        "//src/cdk/testing",
+    ],
+)
+
+ng_test_library(
+    name = "shared_unit_tests",
```
Comment (devversion) [inline src/material/autocomplete/testing/BUILD.bazel]: Optional: I'd personally like to name these things `harness_tests_lib`. Lib already implies that it just exposes the files and using `harness` in the name makes it less confusing (confusion between unit tests of testing and unit tests running through harnesses)

**Comment by mmalerba** [comment]
comments addressed

**Pull request merged**